packages/gatsby/src/services/calculate-dirty-queries.ts

```diff
@@ -0,0 +1,19 @@
+import { calcInitialDirtyQueryIds, groupQueryIds } from "../query"
+import { IBuildContext } from "./"
+
+export interface IGroupedQueryIds {
+  pageQueryIds: string[]
+  staticQueryIds: string[]
+}
+
+export async function calculateDirtyQueries({
+  store,
+}: Partial<IBuildContext>): Promise<{
+  queryIds: IGroupedQueryIds
+}> {
+  const state = store?.getState()
```

Comment:
Does it make sense to do a `?.` here? store will never be nullable. I Feel like typings are wrong

Reply:
I'll replace it with a null check. It's nullable because the context type needs it nullable, as it doesn't exist when setting up the machine

Reply:
oh nvm than, find it a bit odd that it's not set by default

Reply:
I would rather see something like `{store: IBuildContext["store"]}`

Reply:
Yes, it certainly should always be set when in this state. The issue is that the context needs to be the same type for every service. If I make it required then it would need to be required everywhere. We can't pass it in in initial context because loading it has side effects, so right now it's assigned to context as part of bootstrap.